### ISSUE TITLE
Cache hit occurred on the primary key setup-java-Linux-maven not saving cache

### ISSUE THREAD

**Issue by carlosthe19916** [actions/setup-java#362]
**Description:**
The *Post Setup Java" Phase is failing to create an appropriate cache due to:

```
Cache hit occurred on the primary key setup-java-Linux-maven-xxx, not saving cache.
```

**Task version:**
v3

**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Repro steps:**  
A description with steps to reproduce the issue. If your have a public example or repo to share, please provide the link.

Setup a Java using:
```
- name: Setup Java
  uses: actions/setup-java@v3
  with:
    distribution: "temurin"
    java-version: 11
    cache: "maven"
    server-id: ossrh
    server-username: MAVEN_USERNAME
    server-password: MAVEN_PASSWORD
    gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
    gpg-passphrase: MAVEN_GPG_PASSPHRASE
```

I have run my workflows multiple times in the past and never had any error, but unexpectedly today I started seeing the error. You can see the log at https://github.com/windup/windup-pipelines/runs/7742279699?check_suite_focus=true which contains the text:

```
2022-08-09T07:31:47.2964354Z Post job cleanup.
2022-08-09T07:31:47.4587850Z Post job cleanup.
2022-08-09T07:31:47.6321878Z Removing private key from keychain
2022-08-09T07:31:47.6468455Z Cache hit occurred on the primary key setup-java-Linux-maven-20dfec635c6834cb327074976d0c2dcb80d86aa5898e57d942f4300423af00f6, not saving cache.
2022-08-09T07:31:47.6563675Z Post job cleanup.
```

**Expected behavior:**
Cache should be preserved after the phase "Post Setup Java"

**Actual behavior:**
Cache is not preserved for the next job due to the fact that the "Post Setup Java" phase throws an error that doesn't even make GitHub Action fails but to continue.

**Comment by dmitry-shibanov** [comment]
Hello @carlosthe19916. Thank you for your report. According to the logs it does not throw an error. It states that cache hit primary key that is why it does not save the cache again.

**Comment by dmitry-shibanov** [comment]
Sorry for possible misunderstanding. But as I can see the cache was [restored](https://github.com/windup/windup-pipelines/runs/7742279699?check_suite_focus=true#step:4:306) with the same primary key that is why the action didn't save it again because it was not changed.

**Comment by dmitry-shibanov** [comment]
@carlosthe19916 just a gentle ping

**Comment by carlosthe19916** [comment]
Hi @dmitry-shibanov , my apologies for the delay reply.
I think I understand the reason of the failures in my workload. It seems that even the very first job of the whole workflow starts loading a cache whose primary key seems to belong to the last executed workflow. This is what I got from analyzing my workflow:

- job1: loads primary key "AAA" and tries to save the cache using the same primary key "AAA" - *here is the problem*
- job2: will load primary key "AAA" and will create another new cache "BBB" - *expected behavior*
- job3: will load primary key "BBB" and will create another new cache "CCC" - *expected behavior*

If I am right I wonder if having a parameter in the `setup-java` action to indicate to start with a clear cache in the first job of the workflow is a good idea. 

As a side-note: Because I was in hurry, I decided to replace the out of the box cache management of `setup-java (cache: "maven")` and use `actions/cache@v3` to manage the cache keys myself to be sure the first job of the workflow always start empty.
